packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx

```diff
@@ -199,9 +203,146 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
         });
     }
 
+    /**
+     * Find matches for the given editor.
+     * @param searchTerm the search term.
+     * @param widget the editor widget.
+     * @param searchOptions the search options to apply.
+     *
+     * @returns the list of matches.
+     */
+    protected findMatches(searchTerm: string, widget: EditorWidget, searchOptions: SearchInWorkspaceOptions): SearchMatch[] {
+        if (!widget.editor.document.findMatches) {
+            return [];
+        }
+        const results: FindMatch[] = widget.editor.document.findMatches({
+            searchString: searchTerm,
+            isRegex: !!searchOptions.useRegExp,
+            matchCase: !!searchOptions.matchCase,
+            matchWholeWord: !!searchOptions.matchWholeWord,
+            limitResultCount: searchOptions.maxResults
+        });
+
+        const matches: SearchMatch[] = [];
+        results.forEach(r => {
+            const lineText: string = widget.editor.document.getLineContent(r.range.start.line);
+            matches.push({
+                line: r.range.start.line,
+                character: r.range.start.character,
+                length: r.range.end.character - r.range.start.character,
+                lineText
+            });
+        });
+
+        return matches;
+    }
+
+    /**
+     * Convert a pattern to match all directories.
+     * @param pattern the pattern to be converted.
+     */
+    protected convertPatternToGlob(pattern: string): string {
+        // The leading to make the pattern matches in all directories.
+        const globalPrefix = '**/';
+        if (pattern.startsWith(globalPrefix)) {
+            return pattern;
+        }
+        if (pattern.startsWith('./')) {
+            return pattern.replace('./', globalPrefix);
+        }
```

Comment:
This looks odd. Should we replace `./` by the possible workspace roots rather?

Reply:
Yes, I updated the PR. Now it's appending the workspace root URI before the pattern in case of a leading `./`